logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/handler/DeleteMessageHandler.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.sync.receiver.handler
+
+import com.wire.kalium.logic.data.asset.AssetRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.functional.onSuccess
+
+internal interface DeleteMessageHandler {
+    suspend operator fun invoke(
+        content: MessageContent.DeleteMessage,
+        conversationId: ConversationId,
+        senderUserId: UserId
+    )
+}
+
+internal class DeleteMessageHandlerImpl internal constructor(
+    private val messageRepository: MessageRepository,
+    private val assetRepository: AssetRepository
+) : DeleteMessageHandler {
+    override suspend fun invoke(
+        content: MessageContent.DeleteMessage,
+        conversationId: ConversationId,
+        senderUserId: UserId
+    ) {
+        messageRepository.getMessageById(conversationId, content.messageId).onSuccess { messageToRemove ->
+            val canBeDeleted = isSenderVerified(messageToRemove, senderUserId)
```

Comment:
Aren't you missing an exception for for ephemeral messages here? Because they will not be deleted by the original sender but by the user whose timer first runs out.

Reply:
my bad i will change it to delete with no check if it is Ephemeral and self was the sender

for everyone who is not sender then only delete by orinal sender